lib/vtls/schannel.c

```diff
@@ -1665,7 +1665,12 @@ add_cert_to_certinfo(const CERT_CONTEXT *ccert_context, void *raw_arg)
   if(valid_cert_encoding(ccert_context)) {
     const char *beg = (const char *) ccert_context->pbCertEncoded;
     const char *end = beg + ccert_context->cbCertEncoded;
-    int insert_index = (args->certs_count - 1) - args->idx;
+    int insert_index = args->idx;
+    /* Windows prior to 11 22H2 returned certificates in reverse order. */
+    if(curlx_verify_windows_version(10, 0, 22621, PLATFORM_WINNT,
+                                    VERSION_LESS_THAN)) {
+      insert_index += args->certs_count - 1;
+    }
```

Comment:
`curlx_verify_windows_version()` looks to be a quite heavy function, making many API calls, even an indirect one, and it can even recurse. Haven't made measurements though. [^1]

Maybe worth caching the version result in a `raw_arg` variable?

(Having said that, this version check call is now made from a lot of low-level places in `schannel.c`.)

[^1]: https://github.com/curl/curl/blob/57e2bb52aaa7d16adef6d3d1c4b75c242148a85e/lib/version_win32.c#L196-L314

Reply:
Yeah I noticed that too. It seemed to be used all over the place. The results probably should be memoized if it takes too long.

Reply:
It seems reasonable to make such an optimization (if feasible) in a separate PR though.